pubsub/gcppubsub/gcppubsub.go

```diff
@@ -210,8 +211,63 @@ func (o *URLOpener) OpenTopicURL(ctx context.Context, u *url.URL) (*pubsub.Topic
 
 // OpenSubscriptionURL opens a pubsub.Subscription based on u.
 func (o *URLOpener) OpenSubscriptionURL(ctx context.Context, u *url.URL) (*pubsub.Subscription, error) {
-	for param := range u.Query() {
-		return nil, fmt.Errorf("open subscription %v: invalid query parameter %q", u, param)
+	// Set subscription options to use defaults
+	o.SubscriptionOptions = SubscriptionOptions{
+		ReceiveMaxHandlers:  recvBatcherOpts.MaxHandlers,
+		ReceiveMinBatchSize: recvBatcherOpts.MinBatchSize,
+		ReceiveMaxBatchSize: recvBatcherOpts.MaxBatchSize,
+		AckMaxHandlers:      ackBatcherOpts.MaxHandlers,
+		AckMinBatchSize:     ackBatcherOpts.MinBatchSize,
+		AckMaxBatchSize:     ackBatcherOpts.MaxBatchSize,
+	}
+
+	for param, value := range u.Query() {
+		switch param {
+		case "receive-max-handlers":
+			maxHandlers, err := queryParameterInt(value)
+			if err != nil {
+				return nil, fmt.Errorf("open subscription %v: invalid query parameter %q: %v", u, param, err)
+			}
+
+			o.SubscriptionOptions.ReceiveMaxHandlers = int(maxHandlers)
+		case "receive-min-batch-size":
+			minBatchSize, err := queryParameterInt(value)
+			if err != nil {
+				return nil, fmt.Errorf("open subscription %v: invalid query parameter %q: %v", u, param, err)
+			}
+
+			o.SubscriptionOptions.ReceiveMinBatchSize = int(minBatchSize)
+		case "receive-max-batch-size":
+			maxBatchSize, err := queryParameterInt(value)
+			if err != nil {
+				return nil, fmt.Errorf("open subscription %v: invalid query parameter %q: %v", u, param, err)
+			}
+
+			o.SubscriptionOptions.ReceiveMaxBatchSize = int(maxBatchSize)
+		case "ack-max-handlers":
+			maxHandlers, err := queryParameterInt(value)
+			if err != nil {
+				return nil, fmt.Errorf("open subscription %v: invalid query parameter %q: %v", u, param, err)
+			}
+
+			o.SubscriptionOptions.AckMaxHandlers = int(maxHandlers)
+		case "ack-min-batch-size":
```

Comment:
I don't think we need to make the ack batchers configurable. They should be driven by the receive batches.